rfcs/0021-ckb-address-format/0021-ckb-address-format.md

```diff
@@ -21,11 +21,12 @@ To generate a CKB address, we firstly encode lock script to bytes array, name *p
 
 There are several methods to convert lock script into payload bytes array. We use 1 byte to identify the payload format.
 
-| format type |                   description                  |
-|:-----------:|------------------------------------------------|
-|  0x01       | short version for locks with popular code_hash |
-|  0x02       | full version with hash_type = "Data"           |
-|  0x04       | full version with hash_type = "Type"           |
+| format type |                   description                        |
+|:-----------:|------------------------------------------------------|
+|  0x01       | short version for locks with popular code_hash       |
+|  0x02       | full version with hash_type = "Data", deprecated     |
+|  0x04       | full version with hash_type = "Type", deprecated     |
+|  0x08       | full version identified the hash_type and vm_version |
```

Comment:
By new design of [`ScriptHashType`](https://github.com/nervosnetwork/rfcs/pull/238/files#diff-acf3e873b6f19e69d762b0e51b14b50afb4f2928fd109418571f19e88b9d95fdR36) the `vm_version` information is encoded in `hash_type`. So the meaning of `0x08` should be adjusted?

Reply:
I'm curious that we won't use format type as bit flags, why we use 1, 2, 4, 8 instead of 1, 2, 3, 4?

Reply:
> By new design of [`ScriptHashType`](https://github.com/nervosnetwork/rfcs/pull/238/files#diff-acf3e873b6f19e69d762b0e51b14b50afb4f2928fd109418571f19e88b9d95fdR36) the `vm_version` information is encoded in `hash_type`. So the meaning of `0x08` should be adjusted?

The `format type` is a field for address encoding, with not conflict with `hash_type` in type script.  In fact, they are two components combined to implement VM selection.

Reply:
> I'm curious that we won't use format type as bit flags, why we use 1, 2, 4, 8 instead of 1, 2, 3, 4?

I don't know, in the past, the format type is `0x01`, `0x02`, `0x04` which is lack of `0x3`. So, to obey the unknow rule, I choose `0x08`.

Reply:
how about use 0x00 for full payload? after 0x02 and 0x04 are deprecated, the table looks more consistent:
```
| format type |                   description                        |
|:-----------:|------------------------------------------------------|
|  0x00       | full version identified the hash_type and vm_version |
|  0x01       | short version for locks with popular code_hash       |
```